doc/member_types.md

```diff
@@ -51,8 +52,9 @@ nyan Type     | C++ Equivalent       | Operators                | Description
 
 Type Modifier | nyan Types | Description
 --------------|------------|-----------------------
-`optional`    | All        | The member can be initialized with the placeholder value `None`.
+`abstract`    | `object`   | Can assign abstract object references.
 `children`    | `object`   | Only descendants of the object can be assigned, not the object itself.
+`optional`    | All        | The member can be initialized with the placeholder value `None`.
```

Comment:
maybe highlight the difference between an object that becomes abstract because one of its members has no value vs optional value which explicitly can have no value assigned with `None`

Reply:
It is explained in section 3.3 in more detail.

> `None` indicates that the member is initialized, but has no regular value.

I also defined an abstract object as 

> An object is **abstract** iff it contains at least one uninitialized member.
This includes inherited members.

in `nyan_specification.md` to avoid confusing "no value assigned" with "`None` value assigned".